schemas/com.shopify/update-product/jsonschema/1-0-0

```diff
@@ -0,0 +1,272 @@
+          "product_id" : {
+            "type" : "integer",
+            "maximum" : 9223372036854775807,
+            "minimum" : 0
+          },
+          "values" : {
+            "type" : "array",
+            "items" : {
+              "type" : "string"
+            }
+          },
+          "position" : {
+            "type" : "integer",
+            "maximum" : 32767,
+            "minimum" : 0
+          }
+        },
+        "additionalProperties" : false
+      }
+    },
+    "id" : {
+      "type" : "integer",
+      "maximum" : 9223372036854775807,
+      "minimum" : 0
+    },
+    "created_at" : {
+      "type" : "string",
+      "format" : "date-time"
+    },
+    "title" : {
+      "type" : "string"
+    },
+    "template_suffix" : {
+      "type" : "null"
+    },
+    "vendor" : {
+      "type" : "string"
+    },
+    "updated_at" : {
+      "type" : "string",
+      "format" : "date-time"
+    },
+    "images" : {
+      "type" : "array",
+      "items" : {
+        "type" : "object",
+        "properties" : {
+          "variant_ids" : {
+            "type" : "array",
+            "items" : {
+              "type" : "integer",
+              "maximum" : 9223372036854775807,
+              "minimum" : 0
+            }
+          },
+          "id" : {
+            "type" : "integer",
+            "maximum" : 9223372036854775807,
+            "minimum" : 0
+          },
+          "src" : {
+            "type" : "string",
+            "format" : "uri"
+          },
+          "product_id" : {
+            "type" : "integer",
+            "maximum" : 9223372036854775807,
+            "minimum" : 0
+          },
+          "position" : {
+            "type" : "integer",
+            "maximum" : 32767,
+            "minimum" : 0
+          },
+          "created_at" : {
+            "type" : "string",
+            "format" : "date-time"
+          },
+          "updated_at" : {
+            "type" : "string",
+            "format" : "date-time"
+          }
+        },
+        "additionalProperties" : false
+      }
+    },
+    "published_at" : {
+      "type" : [ "string", "null" ],
+      "format" : "date-time"
+    },
+    "variants" : {
+      "type" : "array",
+      "items" : {
+        "type" : "object",
+        "properties" : {
+          "weight" : {
+            "type" : "number",
+            "minimum" : 0
+          },
+          "inventory_policy" : {
+            "type" : "string",
+            "pattern" : "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$"
+          },
+          "sku" : {
+            "type" : "string"
+          },
+          "barcode" : {
+            "type" : "string"
+          },
+          "image_id" : {
+            "type" : [ "integer", "null" ],
+            "maximum" : 9223372036854775807,
+            "minimum" : 0
+          },
+          "option1" : {
+            "type" : "string"
+          },
+          "price" : {
+            "type" : "string"
```

Comment:
Is price with currency (`300USD`)?

Reply:
No, but according to the shopify API docs the response is a string instead of a numerical:

`"line_items": [
    {
      "id": 866550311766439020,
      "variant_id": null,
      "title": "IPod Nano - 8GB",
      "quantity": 1,
      "price": "199.00",
      "grams": 567,
      "sku": "IPOD2008PINK",
      "variant_title": null,
      "vendor": null,
      "fulfillment_service": "manual",
      "product_id": 632910392,
      "requires_shipping": true,
      "taxable": true,
      "gift_card": false,
      "name": "IPod Nano - 8GB",
      "variant_inventory_management": null,
      "properties": [
      ],
      "product_exists": true,
      "fulfillable_quantity": 1,
      "total_discount": "0.00",
      "fulfillment_status": null,
      "tax_lines": [
      ]
    }]`

Reply:
We can have it as `number` then. Thing is that we're writing JSON Schemas for adapter's output (part of Scala Common Enrich), not input. Which means we can programmatically "cast" these values into something more appropriate.

Reply:
Ok, so how do i force a field to go with a given type? Or can schema_guru detect "number" from strings always containing numbers?

Reply:
As with other properties - Schema Guru cannot determine if we've already defined some meaning for them. The only way is to keep editing it manually after each derivation.

I created a ticked to explore it (https://github.com/snowplow/schema-guru/issues/175) as it looks like this use case arises more and more frequent.